### PR TITLE
Use the WASDK self-contained by default

### PR DESCRIPTION
**Pull request by mattleibow**
### Description of Change

This PR adds the fully self-contained support for unpackaged Maui apps. In WASDK 1.0, unpackaged support was added, but this involved the developer making sure the various Windows App SDK runtime MSIX files were installed on the end-user's machine. In 1.1, the WinUI team added support for a fully self contained apps with all the required bits in the box.

See more info on the pros and cons of all the options in this table: https://docs.microsoft.com/en-us/windows/apps/package-and-deploy/deploy-overview

For maui apps, the defaults after this PR will be:
 - MSIX apps will not include the runtime bits as this is usually managed by the store and the OS
 - Unpackaged apps will include the runtime bits to make a fully xcopy to production

All these options can still be controlled with:

 - `WindowsPackageType`: `MSIX` or `None` to control packaged and unpackaged respectively
 - `WindowsAppSDKSelfContained`: `true` or `false` to control whether the app includes the WASDK runtime or not

### Issues Fixed

 - Part of #3166 

Contains the workarounds for https://github.com/microsoft/WindowsAppSDK/issues/2684

### CODE REVIEW

File: .nuspec/Microsoft.Maui.Controls.SingleProject.props [hunk 1]
```diff
@@ -3,6 +3,7 @@
   <PropertyGroup Condition=" '$(SingleProject)' == 'true' and '$([MSBuild]::GetTargetPlatformIdentifier($(TargetFramework)))' == 'windows' ">
     <OutputType Condition="'$(OutputType)' == 'Exe'">WinExe</OutputType>
     <WindowsPackageType Condition=" '$(WindowsPackageType)' == '' and '$(EnableMsixTooling)' == 'true' and ('$(OutputType)' == 'Exe' or '$(OutputType)' == 'WinExe') ">MSIX</WindowsPackageType>
+    <WindowsAppSDKSelfContained Condition=" '$(WindowsAppSDKSelfContained)' == '' and '$(WindowsPackageType)' == 'None' and ('$(OutputType)' == 'Exe' or '$(OutputType)' == 'WinExe') ">true</WindowsAppSDKSelfContained>
```
Comment (eerhardt) [inline .nuspec/Microsoft.Maui.Controls.SingleProject.props:6]: Can we encapsulate `('$(OutputType)' == 'Exe' or '$(OutputType)' == 'WinExe')` so the condition isn't duplicated on all these lines?

Also, can OutputType ever be `Exe`? On line 4 we say if OutputType is `Exe` we are changing it to `WinExe`.

**Comment by eerhardt** [comment]
Is there any way we can test creating an unpackaged app and running it?

File: src/Workload/Microsoft.Maui.Sdk/Sdk/WinUI.Unpackaged.targets [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    </PropertyGroup>
+    <PropertyGroup Condition="'$(Platform)' == 'AnyCPU'">
+      <NativePlatform>neutral</NativePlatform>
+      <NativePlatform Condition="'$(RuntimeIdentifier)' == 'win10-x86'">x86</NativePlatform>
+      <NativePlatform Condition="'$(RuntimeIdentifier)' == 'win10-x64'">x64</NativePlatform>
+      <NativePlatform Condition="'$(RuntimeIdentifier)' == 'win10-arm'">arm</NativePlatform>
+      <NativePlatform Condition="'$(RuntimeIdentifier)' == 'win10-arm64'">arm64</NativePlatform>
+    </PropertyGroup>
+    <ItemGroup>
+      <MicrosoftWindowsAppSDKMsix Include="$([MSBuild]::NormalizeDirectory('$(MicrosoftWindowsAppSDKPackageDir)','tools\Msix\win10-$(NativePlatform)'))Microsoft.WindowsAppRuntime.?.?.Msix"/>
```
Comment (eerhardt) [inline src/Workload/Microsoft.Maui.Sdk/Sdk/WinUI.Unpackaged.targets:34]: Why is this necessary? It doesn't seem to be related to issue 2684.

Comment (mattleibow) [reply src/Workload/Microsoft.Maui.Sdk/Sdk/WinUI.Unpackaged.targets:34]: This is the other part of the issue. It is actually the more important part to copy the msix contents into the bin.

The whole reason for the NativePlatform is to get the value for this and it currently resolves to win10-AnyCPU so I have to re do the target.

**Comment by mattleibow** [comment]
Yep, for the running.

Pick any maui app or the sample and make the WindowsPackageType=None

And then change MsixPackage to Project in the launchSettings.json.

It should run right out the box.

**Comment by eerhardt** [comment]
> Pick any maui app or the sample and make the WindowsPackageType=None

I meant an automated test that will run on CI

**Comment by mattleibow** [comment]
Yeah, my next comment. Incomming device tests for Windows.
